### PR TITLE
Clarify what is Agent, especially when Supervisor is used

### PR DESCRIPTION
**Pull request by tigrannajaryan**
Contributes to https://github.com/open-telemetry/opamp-spec/issues/114

### CODE REVIEW

File: specification.md [hunk 1]
```diff
@@ -247,6 +247,36 @@ the Agent to the Server and from the Server to the Agent.
...
+      │          │
+      └──────────┘
+```
+
+OpAMP specification does not require using Supervisor and does not define how the
+Supervisor and the Agent communicate.
+
+In the rest of the OpAMP specification the term _Agent_ is used to refer to the entity
+which, implements the client portion of the OpAMP, regardless of whether a Supervisor
+is part of that entity or no.
```
Comment (andykellr) [inline specification.md:278]:
```suggestion
In the rest of the OpAMP specification the term _Agent_ is used to refer to the entity
which implements the client portion of the OpAMP, regardless of whether or not a Supervisor
is part of that entity.
```

File: specification.md [hunk 2]
```diff
@@ -247,6 +247,36 @@ the Agent to the Server and from the Server to the Agent.
...
+           ▼
+      ┌──────────┐
+      │          │
+      │  Agent   │
+      │          │
+      └──────────┘
+```
+
+OpAMP specification does not require using Supervisor and does not define how the
+Supervisor and the Agent communicate.
```
Comment (andykellr) [inline specification.md:274]:
```suggestion
The OpAMP specification does not require using a Supervisor and does not define how the
Supervisor and the Agent communicate.
```

**Review by andykellr — APPROVED** [review]
Very minor grammatical feedback, otherwise good.

File: specification.md (hunk 1 above)
Comment (tigrannajaryan) [reply specification.md]: Fixed.

File: specification.md (hunk 2 above)
Comment (tigrannajaryan) [reply specification.md]: Fixed.

**Pull request merged**